### DISCUSSION TITLE
PDF格式的图片插入正文出现部分内容丢失

### DISCUSSION THREAD

**Discussion by ZhanchiWang** [ustctug/ustcthesis#399] [Q&A] [answered]
各位大佬好，最近在用这个模板写毕业论文，感谢大佬们制作和维护的latex模板，帮了大忙！

遇到一个问题，目前不知道什么原因，想请教一下：
我文章中的图片制作的都是pdf格式（为了让图中标注都是矢量），这些pdf上传到overleaf并且预览的时候都没有问题，但是插入正文并且编译之后会有部分图片出现内容丢失。如下：
预览图：
![image](https://github.com/user-attachments/assets/71569fa9-ab7d-4f87-9239-f7a46e69fdc3)
编译后：
![image](https://github.com/user-attachments/assets/b2de77c1-3609-49ee-805a-4350af6fc98c)

尝试替换成png格式则没有问题，相同位置换成其他的pdf格式图片也没问题，就是某些图片，无论放大缩小，修改命名，都存在问题。

想让大佬们帮忙看看是什么问题，感谢！

**Comment by zepinglee** [comment]
有没有报错或警告信息？另外是不是 overleaf 的预览的问题？建议将编译后的 PDF 下载到本地检查一下。

**Reply by ZhanchiWang** [reply]
感谢回复！
编译的时候没有报错，下载下来的pdf也有错误，并且adobe acrobat在这个页面会报错
![image](https://github.com/user-attachments/assets/f19708e1-1c96-45dd-b2fa-8eeedd43a896)

**Reply by zepinglee** [reply]
那大概率是 xetex 嵌入 PDF 的 bug。如果可以的话，建议将这张 PDF 图片和相关的 LaTeX 代码发上来供测试。

**Reply by ZhanchiWang** [reply]
谢谢！下面是pdf和png格式的图片：
[5.4.pdf](https://github.com/user-attachments/files/16611388/5.4.pdf)

![5 4](https://github.com/user-attachments/assets/329b56c0-a0c3-4d19-9704-8697b7ce92b1)

下面是使用的代码：
```latex
\begin{figure}[h]
  \centering
  \includegraphics[width=\textwidth]{figures/5.4.pdf}
  \caption{机器人动态运动}
  \label{fig:whipping}
  \note{注：图注的内容不宜放到图题中。}
\end{figure}
```

下面这张pdf也是发现会出现问题的：
[2.5.pdf](https://github.com/user-attachments/files/16611431/2.5.pdf)

**Reply by zepinglee** [reply, marked as answer]
使用 xetex 或 luatex 都能复现这个问题，而 pdftex 没有问题。那应该大概率是前面两个引擎的问题了。这些测试都用的标准文档类，所以这个问题跟模板本身无关。

```latex
\documentclass{article}
\usepackage{graphicx}
\begin{document}
\centering
\includegraphics[width=\textwidth]{5.4.pdf}
\includegraphics[width=\textwidth]{5.4.png}
\end{document}
```

<img width="374" alt="Screenshot 2024-08-15 at 10 12 47" src="https://github.com/user-attachments/assets/2f635616-4182-4230-b6de-528eb10c394b">

**Reply by ZhanchiWang** [reply]
非常感谢！我看能不能给他们反馈一下，我先在文章用png格式来解决一下！再次感谢~

**Reply by ellenalexander** [reply]
我也遇到了同样的问题，我用的是XeLatex, 2026年了还没解决这个BUG……

**Reply by zepinglee** [reply]
> 我也遇到了同样的问题，我用的是XeLatex, 2026年了还没解决这个BUG……

建议发到 http://tex.stackexchange.com/ 问一下。记得附上 MRE 和 PDF 图片。
